src/components/button-toggle/button-toggle.pw.tsx

```diff
@@ -0,0 +1,858 @@
+      "true"
+    );
+  });
+
+  ([
+    [25, 341],
+    [50, 683],
+    [100, 1366],
+  ] as [ButtonToggleGroupProps["inputWidth"], number][]).forEach(
+    ([labelWidth, width]) => {
+      test(`should render Button-Toggle-Group with labelWidth prop of ${labelWidth} and width of ${width}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<ButtonToggleGroupNotInBox inputWidth={labelWidth} />);
+
+        const buttonParent = buttonToggleButton(page)
+          .locator("..")
+          .locator("..");
+        await assertCssValueIsApproximately(buttonParent, "width", width);
+      });
+    }
+  );
+
+  test(`should render Button-Toggle-Group with helpAriaLabel set to playwright_data`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ButtonToggleGroupComponent helpAriaLabel={testPropValue} />);
+
+    await expect(buttonToggleGroupHelpIcon(page)).toHaveAttribute(
+      "aria-label",
+      testPropValue
+    );
+  });
+
+  ([
+    ["8px", 1],
+    ["16px", 2],
+  ] as const).forEach(([padding, spacing]) => {
+    test(`should render Button-Toggle-Group with padding of ${padding} if labelSpacing prop is ${spacing}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <ButtonToggleGroupComponent labelInline labelSpacing={spacing} />
+      );
+
+      await expect(
+        getDataElementByValue(page, "label").locator("..")
+      ).toHaveCSS("padding-right", padding);
+    });
+  });
+
+  test(`container div should auto flex`, async ({ mount, page }) => {
+    await mount(<ButtonToggleGroupComponent fullWidth />);
+
+    await expect(buttonTogglePreview(page).nth(0)).toHaveCSS(
+      "flex",
+      "1 1 auto"
+    );
+  });
+
+  // actual width found now 456px - test should be reviewed
+  test(`width of label should be 100% / 450px`, async ({ mount, page }) => {
+    await mount(<ButtonToggleGroupNotInBox fullWidth />);
+
+    await assertCssValueIsApproximately(
+      buttonToggleButton(page).nth(0),
+      "width",
+      456
+    );
+  });
+});
+
+test.describe("Event tests for group component", () => {
+  test(`should call onChange callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ButtonToggleGroupComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await buttonTogglePreview(page).nth(0).click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onChange callback with undefined when a click event is triggered on the currently-selected button and the allowDeselect prop is true`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ButtonToggleGroupComponent
+        value="foo"
+        allowDeselect
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await buttonTogglePreview(page).nth(0).click();
+    expect(callbackCount).toBe(1);
```

Comment:
**question/suggestion(non-blocking):** The test description states that `onChange` callback should be called with `undefined` but here we are just checking the `callbackCount` has increased? I'm not sure if we can see what the `onChange` returns in PW though.